miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -7,29 +7,67 @@ import com.rakuten.tech.mobile.miniapp.display.Displayer
 import com.rakuten.tech.mobile.miniapp.js.MiniAppMessageBridge
 import com.rakuten.tech.mobile.miniapp.navigator.MiniAppNavigator
 import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermission
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionType
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionResult
 import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionCache
+import com.rakuten.tech.mobile.miniapp.storage.MiniAppManifestCache
 
-@Suppress("TooManyFunctions")
+@Suppress("TooManyFunctions", "LongMethod", "ExpressionBodySyntax")
 internal class RealMiniApp(
     private val apiClientRepository: ApiClientRepository,
     private val miniAppDownloader: MiniAppDownloader,
     private val displayer: Displayer,
     private val miniAppInfoFetcher: MiniAppInfoFetcher,
-    private val miniAppCustomPermissionCache: MiniAppCustomPermissionCache
+    private val miniAppCustomPermissionCache: MiniAppCustomPermissionCache,
+    private val miniAppManifestCache: MiniAppManifestCache
 ) : MiniApp() {
 
+    @VisibleForTesting
+    var temporaryManifest: MiniAppManifest? = null
+
     override suspend fun listMiniApp(): List<MiniAppInfo> = miniAppInfoFetcher.fetchMiniAppList()
 
     override suspend fun fetchInfo(appId: String): MiniAppInfo = when {
         appId.isBlank() -> throw sdkExceptionForInvalidArguments()
         else -> miniAppInfoFetcher.getInfo(appId)
     }
 
-    override fun getCustomPermissions(miniAppId: String): MiniAppCustomPermission =
-        miniAppCustomPermissionCache.readPermissions(miniAppId)
+    override fun getCustomPermissions(
+        miniAppId: String
+    ): MiniAppCustomPermission {
+        // return only the permissions listed in the Mini App's manifest.
+        val manifestPermissions = miniAppManifestCache.getCachedAllPermissions(miniAppId)
+        return MiniAppCustomPermission(miniAppId, manifestPermissions)
+    }
+
+    override fun setCustomPermissions(miniAppCustomPermission: MiniAppCustomPermission) {
```

Comment:
I think it's better to remove the `temporaryManifest` and change the signature into `setCustomPermission(miniAppManifest, miniAppCustomPermission)`

Reply:
Yeah, we could. But, it will be extra work for the hostapp as SDK's miniapp instance already have the manifest information, so HostApp doesn't need to send it additionally to store the permission.

Reply:
I'm thinking about user does not know which miniapp they are set custom permission. When they use it wrongly, they will think it's sdk fault.

I'll let you & @corycaywood make a decision on it 😄

Reply:
Ok, I found a discussion in iOS side: https://github.com/rakutentech/ios-miniapp/pull/227#discussion_r587246181
I'll follow up accordingly. I'll vote for caching it right after fetching from api, so we don't need this `temporaryManifest` variable anymore.

Reply:
Actually the discussion about caching the manifest was relevant to checking the manifest permissions when calling `MiniApp.create`, not when calling `MiniApp.setCustomPermissions`. This is a little bit tricky, so I'm not sure how to handle it...

The `temporaryManifest` variable here is just caching the last downloaded manifest without saving the mini app id or version, right? So how do you know that you're checking the correct manifest? Also, `setCustomPermissions` can be used on the permissions management screen, so I think this current implementation could cause problems in that use case.

Maybe you don't need to check the manifest permissions when calling `setCustomPermission`? i.e. we allow the host app to store any permissions, even if not in the manifest. But then we would need to change some of our implementation so that we always check the manifest when reading the cached permissions. What do you think?

Reply:
Okay, I have updated this part.

Yeah, it makes sense not to check manifest permissions when calling `setCustomPermission`. But, we still need to check manifest when calling `getCustomPermissions`... 🤔 Also, I have removed `temporaryManifest` and saving the manifest when fetching the latest manifest from api. 

@corycaywood Do you think, it's looked okay, then I'll fix the broken tests.

Reply:
Sorry, I had another comment that I forgot to click the submit button on 😣 

Maybe we could store all of the passed permissions when `setCustomPermissions` is called.

An then when the Mini App is launched (`MiniApp.create`), we can do the following:
- Check if manifest is downloaded (or cached in memory) for the specified mini app id and version
  - If not, then download the manifest
- Check the cached permissions and update them to remove any permissions which aren't contained in the manifest

What do you think?

Reply:
Updated.